lang/cs/Org.Apache.REEF.Wake/Time/Time.cs

```diff
@@ -50,33 +55,21 @@ public override bool Equals(object obj)
             {
                 return true;
             }
-            Time other = obj as Time;
+            
+            // Note: This allows for the very strange semantic where
+            // two different subclasses might return true for Equals.
+            var other = obj as Time;
```

Comment:
I agree with you here, but it seems that different types of `Time` need to be compared with each other in the `RuntimeClock`, hence the original workaround.